libraries/AP_DDS/AP_DDS_External_Odom.cpp

```diff
@@ -0,0 +1,60 @@
+
+
+#if AP_DDS_ENABLED && defined(HAL_VISUALODOM_ENABLED)
+
+#include "AP_DDS_External_Odom.h"
+#include <AP_VisualOdom/AP_VisualOdom.h>
+
+void AP_DDS_External_Odom::handle_external_odom(const tf2_msgs_msg_TFMessage& msg)
+{
+    AP_VisualOdom *visual_odom = AP::visualodom();
+    if (visual_odom != nullptr) {
+        for (size_t i = 0; i < msg.transforms_size; i++) {
+            const auto ros_transform_stamped = msg.transforms[i];
+            if (is_odometry_frame(ros_transform_stamped)) {
+                const uint64_t remote_time_us {0}; // TODO
+                const uint32_t time_ms {0}; // TODO
```

Comment:
This is populated by the jitter correction with local millis since boot. EKF uses time since boot. Unix time would require uint64_t. 

[GCS_Mavlink.cpp ](https://github.com/ArduPilot/ardupilot/blob/3fd08865745fb651604f746135d8b3f9ea0bcf70/libraries/GCS_MAVLink/GCS_Common.cpp#L3569)

Reply:
Will do later